javascriptv3/example_code/bedrock-runtime/config/foundation_models.js

```diff
@@ -0,0 +1,83 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+export const FoundationModels = Object.freeze({
+  CLAUDE_3_HAIKU: {
+    modelId: "anthropic.claude-3-haiku-20240307-v1:0",
+    modelName: "Anthropic Claude 3 Haiku",
+    module: () => import("../models/anthropic_claude/claude_3.js"),
+    invoker: (/** @type {Function} */ module) => module.invokeModel,
```

Comment:
This typing doesn't look right. Since it looks like every `invokeModel` behaves the same, you could do something like this:
```
/**
 * @typedef {(prompt: string, modelId: string) => Promise<string[]>} Invoker
 */

/**
 * @typedef {{ invokeModel: Invoker }} Model
 */

export const FoundationModels = Object.freeze({
  CLAUDE_3_HAIKU: {
    modelId: "anthropic.claude-3-haiku-20240307-v1:0",
    modelName: "Anthropic Claude 3 Haiku",
    module: () => import("../models/anthropic_claude/claude_3.js"),
    invoker: (/** @type Model **/ model) => model.invokeModel,
  },
```

Reply:
Unfortunately they don't behave all the same. Some models return an array of responses, some only a single response.

Reply:
Understood, but the modules could.

Reply:
Not a deal breaker. I'll resolve the convo and you can leave it as is if you like.